libs/use-viewer/src/index.ts

```diff
@@ -11,16 +11,16 @@ import {
 import { useReducer, useRef, useState } from 'react';
 
 import reducer from './reducer';
-import { RemoteTrackSources, SimulcastQuality, SourceId, Viewer, ViewerActionType, ViewerProps } from './types';
+import { RemoteTrackSources, SimulcastQuality, Viewer, ViewerActionType, ViewerProps } from './types';
 import { addRemoteTrackAndProject, unprojectFromStream, projectToStream } from './utils';
 
-const DEFAULT_SOURCE_ID = new Date().valueOf().toString();
-const INITIAL_REMOTE_TRACK_SOURCES = new Map() as RemoteTrackSources;
-
 const useViewer = ({ handleError, streamAccountId, streamName, subscriberToken }: ViewerProps): Viewer => {
   const viewerRef = useRef<View>();
 
-  const [remoteTrackSources, dispatch] = useReducer(reducer, INITIAL_REMOTE_TRACK_SOURCES);
+  const [remoteTrackSources, dispatch] = useReducer(reducer, new Map() as RemoteTrackSources);
+
+  const remoteTrackSourcesRef = useRef<RemoteTrackSources>();
```

Comment:
Use https://www.npmjs.com/package/react-usestateref instead of creating a new ref

Reply:
I've checked the source for that npm package, it's specific to `useState` and not `useReducer`.